src/task/Poller.ts

```diff
@@ -5,25 +5,26 @@ interface PollerOptions {
   concurrency: number;
 }
 
-export class Poller {
-  private concurrentCalls: Array<{
-    promise: Promise<void>;
-    stop: () => Promise<boolean>;
-  }> = [];
-  private pollFunction: () => Promise<void> = async () => {};
+export class Poller<T> {
+  private timeoutHandler?: NodeJS.Timeout;
+  private pollFunction: (count: number) => Promise<T[]>;
+  private performWorkFunction: (work: T) => Promise<void> = async () => {};
   private polling = false;
+  private _tasksInProcess = 0;
   options: PollerOptions = {
     pollInterval: 1000,
```

Comment:
This might be too low --- other SDKs make this at default to 100.  Is this in millis?

Reply:
yes its millis